packages/langium/src/parser/langium-parser.ts

```diff
@@ -247,12 +248,19 @@ export class LangiumParser {
             this.stack.pop();
         }
         if (obj.$type === DatatypeSymbol) {
-            const node = obj.$cstNode;
-            return node.text;
+            return this.constructDataTypeResult(obj.$cstNode);
         }
         return obj;
     }
 
+    private constructDataTypeResult(node: CstNode): string {
+        let text = '';
+        for (const subNode of streamCst(node).filter(isLeafCstNode)) {
```

Comment:
Shall we remove hidden tokens here? For example, a sequence `a .b .c` should result in the internal string value `a.b.c` in a language with qualified names.

Reply:
Right, it also filters for hidden tokens now.